cli/crates/gateway/src/serving.rs

```diff
@@ -101,7 +101,7 @@ async fn get_graphql(
     let ctx = crate::Context::new(headers, sender);
 
     let mut request: engine::Request = params.into();
-    request.ray_id = ctx.ray_id.clone();
+    request.ray_id.clone_from(&ctx.ray_id);
```

Comment:
I find the `.clone_from()` and `.clone_into()` less readable than assignments. Does clippy give good reasons for enforcing that?

Reply:
```
Performs copy-assignment from source.

a.clone_from(&b) is equivalent to a = b.clone() in functionality, but can be overridden to reuse the resources of a to avoid unnecessary allocations.
```
I'm assuming they added a check for this since it can be more performant

Reply:
These are mostly auto-fixed by the way

Reply:
I thought the same tbh

Reply:
We could bypass this check everywhere, although I'm not sure these few lines warrant it

Reply:
@obmarg @tomhoule any preference?

Reply:
Hmm yes so it's about specialization. I still find it less readable but I get the point now. No strong opinion.

Reply:
I suppose I can live with it.